enclave/core/sgx/calls.c

```diff
@@ -279,15 +280,57 @@ oe_result_t oe_handle_call_enclave_function(uint64_t arg_in)
         args.output_buffer_size,
         &output_bytes_written);
 
-    // The output_buffer is expected to point to a marshaling struct,
-    // whose first field is an oe_result_t. The function is expected
-    // to fill this field with the status of the ecall.
-    result = *(oe_result_t*)output_buffer;
+    /*
+     * The output_buffer is expected to point to a marshaling struct.
+     * The function is expected to fill the struct.
+     */
+    return_args_ptr = (oe_call_function_return_args_t*)output_buffer;
 
+    result = return_args_ptr->result;
     if (result == OE_OK)
     {
+        /*
+         * Error out the case if the deepcopy_out_buffer is NULL but the
+         * deepcopy_out_buffer_size is not zero.
```

Comment:
What about the case of buffer non-NULL and size 0, should that be an error?

Reply:
Added check